### PR TITLE
feat: toHaveAccessibilityValue() matcher

### PR DESCRIPTION
**Pull request by mdjastrzebski**
**What**:

Matcher for matching accessibility value.

**Why**:

Current `toHaveProp` is performing strict matching of `accessibilityValue` prop, while matching based only on some a11y value entries is useful.

**How**:

Matching code based on `*ByA11yValue` matcher.

**Checklist**:

- [x] Documentation added to the [docs](https://github.com/testing-library/jest-native/README.md)
- [x] Typescript definitions updated
- [x] Tests
- [x] Ready to be merged 

@pierrezimmermannbam, @AugustinLF, @MattAgn could you help with reviewing this?

### CODE REVIEW

**Review by AugustinLF — APPROVED** [review]
Seems good to me. Might be nice to have a second review, I'm not as familiar with this lib as with RNTL.

**Comment by mdjastrzebski** [comment]
@pierrezimmermannbam There is indeed some code duplication between RNTL and JN. But there are also parts when the they have same concepts but different behaviour which is even worse. I share you point that about bringing them closer together in some form e.g. monorepo or including them in our package, etc. There is definitely benefit in RNTL users using JN matchers, as some assertions are better expressed through queries and some through Jest matchers.

**Pull request merged**
